src/core/modules/Planet.cpp

```diff
@@ -844,6 +844,108 @@ QString Planet::getInfoStringPeriods(const StelCore *core, const InfoStringGroup
 	return str;
 }
 
+class SolarEclipse
+{
+private:
+	double raSun = 0, deSun = 0, sdistanceAu = 0, raMoon = 0, deMoon = 0, mdistanceER = 0, gast = 0;
+	double rss = 0, a = 0, b = 0, d = 0, x = 0, y = 0, z = 0, mu = 0, f1 = 0, f2 = 0, tf1 = 0, tf2 = 0;
+	double L1 = 0, L2 = 0, lat = 0, lon = 0, mag = 0, cd = 0, rho1 = 0;
+	double y1 = 0, xi = 0, eta1 = 0, sd1 = 0, cd1 = 0, rho2 = 0, sd = 0, sd1d2 = 0, cd1d2 = 0;
+	double zeta1 = 0, zeta = 0, sd2 = 0, theta = 0, sfn1 = 0, cfn1 = 0;
+	const double SunEarth = 109.12278; // ratio of Sun-Earth radius 696000/6378.1366
+
+public:
+	Vec3d point(const StelCore* core1)
+	{
+		static SolarSystem* ssystem = GETSTELMODULE(SolarSystem);
+		StelUtils::rectToSphe(&raSun, &deSun, ssystem->getSun()->getEquinoxEquatorialPos(core1));
+		StelUtils::rectToSphe(&raMoon, &deMoon, ssystem->getMoon()->getEquinoxEquatorialPos(core1));
+
+		// R.A. of the Sun
```

Comment:
It may be better to keep the angles in radians, instead of finding degrees here and re-converting to radians below.